truss-train/truss_train/deployment.py

```diff
@@ -29,7 +32,9 @@ def model_dump(self, *args, **kwargs):
 
 def prepare_push(api: BasetenApi, config: pathlib.Path, training_job: TrainingJob):
     # Assume config is at the root of the directory.
-    archive = archive_dir(config.absolute().parent)
+    archive = archive_dir(
```

Comment:
Instead of passing this size_limit_mb down, since we're the only ones that need it now, could you do `os.path.getsize(archive.name)` after the packing? From below it doesn't look like we bail early once we pass the threshold, so no perf impact

Then we can keep the error handling / math localized to this file for now